test/test_io2.c

```diff
@@ -142,7 +142,7 @@ write_invalid_files (p4est_t * p4est)
 
     /* invalid3 */
     snprintf (string0, P4EST_NUM_METADATA_BYTES + 1,
-              "%.7s\n%-23s\n%-15s\n%.16ld", P4EST_MAGIC_NUMBER,
```

Comment:
We could use a stringify macro to make it `"first string" SC_TOSTRING (NUM_BYTES_XYZ) "second string"`.

Reply:
Do you prefer the usage of the stringify macro? If I understand your proposed usage of `SC_TOSTRING` correct `snprintf` has the advantage of being explicit on the number of character that are used express the different number of bytes.

Reply:
We can keep the `snprintf` format strings, but should keep a note that when changing any element lengths, the code needs to be updated. The stringify option would replace 7, 23, 15, 16 by evaluating macros to strings for the same result, with future-proof but more difficult to understand code.

Reply:
Let's call the snprintf format strings good for today and take a note to reconsider them, at the same time that we think about explicit byte counts in the documentation.